posthog/hogql_queries/lifecycle_hogql_query.py

```diff
@@ -0,0 +1,179 @@
+    """
+    )
+
+    return time_filter, date_from, date_to
+
+
+def create_events_query(interval: str, event_filter: ast.Expr):
+    one_interval_period = parse_expr(f"toInterval{interval.capitalize()}(1)")
+
+    if not event_filter:
+        event_filter = ast.Constant(value=True)
+
+    placeholders = {
+        "event_filter": event_filter,
+        "interval": ast.Constant(value=interval),
+        "one_interval_period": one_interval_period,
+    }
+
+    events_query = parse_select(
+        """
+            SELECT
+                events.person.id as person_id,
+                min(events.person.created_at) AS created_at,
+                arraySort(groupUniqArray(dateTrunc({interval}, events.timestamp))) AS all_activity,
+                arrayPopBack(arrayPushFront(all_activity, dateTrunc({interval}, created_at))) as previous_activity,
+                arrayPopFront(arrayPushBack(all_activity, dateTrunc({interval}, toDateTime('1970-01-01 00:00:00')))) as following_activity,
+                arrayMap((previous, current, index) -> (previous = current ? 'new' : ((current - {one_interval_period}) = previous AND index != 1) ? 'returning' : 'resurrecting'), previous_activity, all_activity, arrayEnumerate(all_activity)) as initial_status,
+                arrayMap((current, next) -> (current + {one_interval_period} = next ? '' : 'dormant'), all_activity, following_activity) as dormant_status,
+                arrayMap(x -> x + {one_interval_period}, arrayFilter((current, is_dormant) -> is_dormant = 'dormant', all_activity, dormant_status)) as dormant_periods,
+                arrayMap(x -> 'dormant', dormant_periods) as dormant_label,
+                arrayConcat(arrayZip(all_activity, initial_status), arrayZip(dormant_periods, dormant_label)) as temp_concat,
+                arrayJoin(temp_concat) as period_status_pairs,
+                period_status_pairs.1 as start_of_period,
+                period_status_pairs.2 as status
+            FROM events
+            WHERE {event_filter}
+            GROUP BY person_id
+        """,
+        placeholders=placeholders,
+    )
+    return events_query
+
+
+def run_lifecycle_query(
+    team: Team,
+    query: LifecycleQuery,
+) -> Dict[str, Any]:
+    now_dt = datetime.now()
+
+    try:
+        interval = query.interval.name
+    except AttributeError:
+        interval = "day"
+    if interval not in ["minute", "hour", "day", "week", "month", "quarter", "year"]:
+        raise ValueError(f"Invalid interval: {interval}")
+    one_interval_period = parse_expr(f"toInterval{interval.capitalize()}(1)")
+    number_interval_period = parse_expr(f"toInterval{interval.capitalize()}(number)")
+
+    query_date_range = QueryDateRange(date_range=query.dateRange, team=team, interval=query.interval, now=now_dt)
+
+    time_filter, date_from, date_to = create_time_filter(query_date_range, interval=interval)
+    event_filter = time_filter  # TODO: add all other filters
```

Comment:
This seems like a todo. I think we're still missing property filters (implemented for EventsQuery) and the test account filters (could be built on top of that implementation).